tests/src/storage_string/tests/functions/store_string.rs

```diff
@@ -0,0 +1,68 @@
+    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Nullam feugiat, turpis at pulvinar vulputate, erat libero tristique tellus, nec bibendum odio risus sit amet ante. Aliquam erat volutpat. Nunc auctor. Mauris pretium quam et urna. Fusce nibh. Duis risus. Curabitur sagittis hendrerit ante. Aliquam erat volutpat. Vestibulum erat nulla, ullamcorper nec, rutrum non, nonummy ac, erat. Duis condimentum augue id magna semper rutrum. Nullam justo enim, consectetuer nec, ullamcorper ac, vestibulum in, elit. Proin pede metus, vulputate nec, fermentum fringilla, vehicula vitae, justo. Fusce consectetuer risus a nunc. Aliquam ornare wisi eu metus. Integer pellentesque quam vel velit. Duis pulvinar.
+    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi gravida libero nec velit. Morbi scelerisque luctus velit. Etiam dui sem, fermentum vitae, sagittis id, malesuada in, quam. Proin mattis lacinia justo. Vestibulum facilisis auctor urna. Aliquam in lorem sit amet leo accumsan lacinia. Integer rutrum, orci vestibulum ullamcorper ultricies, lacus quam ultricies odio, vitae placerat pede sem sit amet enim. Phasellus et lorem id felis nonummy placerat. Fusce dui leo, imperdiet in, aliquam sit amet, feugiat eu, orci. Aenean vel massa quis mauris vehicula lacinia. Quisque tincidunt scelerisque libero. Maecenas libero. Etiam dictum tincidunt diam. Donec ipsum massa, ullamcorper in, auctor et, scelerisque sed, est. Suspendisse nisl. Sed convallis magna eu sem. Cras pede libero, dapibus nec, pretium";
+    let input = String {
+        bytes: Bytes(string.as_bytes().to_vec()),
+    };
+
+    assert_eq!(stored_len(&instance).await, 0);
+
+    store_string(input.clone(), &instance).await;
+
+    assert_eq!(stored_len(&instance).await, string.as_bytes().len() as u64);
+    assert_eq!(get_string(&instance).await, input.bytes);
+}
+
+#[tokio::test]
+async fn stores_string_twice() {
+    let instance = setup().await;
+
+    let string1 = "Fuel is the fastest modular execution layer";
+    let string2 = "Fuel is blazingly fast!";
+    let input1 = String {
+        bytes: Bytes(string1.as_bytes().to_vec()),
+    };
+    let input2 = String {
+        bytes: Bytes(string2.as_bytes().to_vec()),
+    };
+
+    store_string(input1.clone(), &instance).await;
+
+    assert_eq!(get_string(&instance).await, input1.bytes);
+    assert_eq!(stored_len(&instance).await, string1.as_bytes().len() as u64);
+
+    store_string(input2.clone(), &instance).await;
+
+    assert_eq!(get_string(&instance).await, input2.bytes);
+    assert_eq!(stored_len(&instance).await, string2.as_bytes().len() as u64);
```

Comment:
I'd add 1 more assertion to prove that the strings are different and storing one after the other results in the previous one being overwritten entirely

Reply:
I'm not sure if I understand. You want me to add
```rust
assert!(string1 != string2);
```
when they are hardcoded?

Reply:
For correctness / completeness of the test we should prove that after the second storage operation the string is different. This test does not enforce this and thus I could make the strings take the same value and everything would pass.

If the test is refactored in any way we need to make sure that something does not cause a false positive and thus we must prove that the strings are different.

Tests should be written as close to the purpose of a function as reasonable i.e. I should be able to pass multiple strings into this and they should all behave as expected e.g. no false positives.